data/transactions/logic/eval.go

```diff
@@ -2102,6 +2103,56 @@ func opGtxnsa(cx *evalContext) {
 	cx.stack[last] = sv
 }
 
+func opGaidImpl(cx *evalContext, groupIdx int, opName string) (sv stackValue, err error) {
+	if groupIdx >= len(cx.TxnGroup) {
+		err = fmt.Errorf("%s lookup TxnGroup[%d] but it only has %d", opName, groupIdx, len(cx.TxnGroup))
+		return
+	} else if groupIdx > cx.GroupIndex {
+		err = fmt.Errorf("%s can't get future creatable ID of txn with index %d", opName, groupIdx)
+		return
+	} else if groupIdx == cx.GroupIndex {
+		err = fmt.Errorf("can't use %s on self, use `global CurrentApplicationID` instead", opName)
+		return
+	} else if txn := cx.TxnGroup[groupIdx].Txn; !(txn.Type == protocol.ApplicationCallTx || txn.Type == protocol.AssetConfigTx) {
+		err = fmt.Errorf("can't use %s on txn that is not an app call nor an asset config txn with index %d", opName, groupIdx)
+		return
+	}
+
+	cid, err := cx.getCreatableID(groupIdx)
```

Comment:
if err!=nil, do you return immediately or do you still want to create sv?

Reply:
My assumption is that an error indicates an application call failure, which causes code execution to end and the transaction/transaction group to be rejected. In other words, the `sv` would never be used if `err != nil`, so I don't see a need to set it.